src/index/base.cpp

```diff
@@ -119,6 +116,16 @@ bool BaseIndex::Init()
             return InitError(strprintf(Untranslated("%s best block of the index goes beyond pruned data. Please disable the index or reindex (which will download the whole blockchain again)"), GetName()));
         }
     }
+
+    // Child init
+    if (!CustomInit(start_block ? std::make_optional(interfaces::BlockKey{start_block->GetBlockHash(), start_block->nHeight}) : std::nullopt)) {
```

Comment:
>  Is my understanding correct that this race cannot actually happen in practice on master right now, because the indexes are set up (Init()) by the init thread, which starts the networking and loadblk threads only later - so I can't see from which other thread BlockConnected signals could come from at this stage.

Yeah, the race cannot realistically happen in master. The only process that occurs before the indexes initialization is the chain state loading process, which doesn't signal any block connection event.

So back porting seems to not be needed. But, if it helps review, could still decouple 21812970cc56191e18f96692e47f00ed881bd596 into a standalone PR.